lib/ecl/ecl_sum.cpp

```diff
@@ -821,35 +804,52 @@ ecl_sum_type * ecl_sum_alloc_resample(const ecl_sum_type * ecl_sum, const char *
   if ( util_string_equal(smspec_node_get_unit(node), "DAYS" ) )
     time_in_days = true;
 
-  ecl_sum_type * ecl_sum_resampled = ecl_sum_alloc_writer( ecl_case , ecl_sum->fmt_case , ecl_sum->unified , ecl_sum->key_join_string , start_time , time_in_days , grid_dims[0] , grid_dims[1] , grid_dims[2] );
-
-
+  //create elc_sum_resampled with TIME node only
+  ecl_sum_type * ecl_sum_resampled = ecl_sum_alloc_writer( ecl_case , ecl_sum->fmt_case , ecl_sum->unified , ecl_sum->key_join_string , input_start , time_in_days , grid_dims[0] , grid_dims[1] , grid_dims[2] );
 
+  //add remaining nodes
   for (int i = 0; i < ecl_smspec_num_nodes(ecl_sum->smspec); i++) {
     const smspec_node_type * node = ecl_smspec_iget_node_w_node_index(ecl_sum->smspec, i);
-    if (util_string_equal(smspec_node_get_gen_key1(node), "TIME"))
-      continue;
-
+    if (util_string_equal(smspec_node_get_keyword(node), "TIME"))
+        continue;
     ecl_sum_add_smspec_node(  ecl_sum_resampled, node );
   }
 
   /*
     The SMSPEC header structure has been completely initialized, it is time to
     start filling it up with data.
+
   */
   ecl_sum_vector_type * ecl_sum_vector = ecl_sum_vector_alloc(ecl_sum, true);
   double_vector_type * data = double_vector_alloc( ecl_sum_vector_get_size(ecl_sum_vector) , 0);
 
-
   for (int report_step = 0; report_step < time_t_vector_size(times); report_step++) {
-    time_t t       = time_t_vector_iget(times, report_step);
-    time_t input_t = time_t_vector_iget(_times, report_step);
-
-    /* Look up interpolated data in the original case. */
-    ecl_sum_get_interp_vector( ecl_sum, t, ecl_sum_vector, data);
+    time_t input_t = time_t_vector_iget(times, report_step);
+    if (input_t < start_time) {
+      //clamping to the first value for t < start_time or if it is a rate than derivative is 0
+      for (int i=1; i < ecl_smspec_num_nodes(ecl_sum->smspec); i++) {
+        double value = 0;
+        const smspec_node_type * node = ecl_smspec_iget_node_w_node_index(ecl_sum->smspec, i);
+        if (!smspec_node_is_rate(node))
+            value = ecl_sum_iget_first_value(ecl_sum,i-1);
```

Comment:
Instead of `i-1` here the code should use `smspec_node_get_params_index(node)` - they can be different.